schemaFile breaks import

The new `schemaFile` syntax added in #455 is very nice but breaks imports.

```
zed import studio-entitlements-admin-server/src/main/resources/authzed/schema.zed.yaml
panic: interface conversion: interface {} is *decode.SchemaRelationships, not *validationfile.ValidationFile

goroutine 1 [running]:
github.com/authzed/zed/internal/decode.unmarshalAsYAMLOrSchemaWithFile({0x1400016e000, 0x3b8e, 0x4000}, {0x101b38340, 0x140000a8640}, {0x16f32a0e3, 0x4b})
        internal/decode/decoder.go:116 +0x224
github.com/authzed/zed/internal/decode.DecoderForURL.fileDecoder.func2({0x101b38340, 0x140000a8640})
        internal/decode/decoder.go:62 +0x9c
github.com/authzed/zed/internal/cmd.importCmdFunc(0x1028c4940, {0x140003c1c50, 0x1, 0x1?})
        internal/cmd/import.go:80 +0xa8
github.com/spf13/cobra.(*Command).execute(0x1028c4940, {0x140003c1c20, 0x1, 0x1})
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0x834
github.com/spf13/cobra.(*Command).ExecuteC(0x140002da008)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1034
github.com/authzed/zed/internal/cmd.Run()
        internal/cmd/cmd.go:134 +0x7a8
main.main()
        cmd/zed/main.go:6 +0x1c
```

I suppose composable schema support would also fix this but just adding it in case it's something quick to support :)

--- a/internal/cmd/import.go
+++ b/internal/cmd/import.go
@@ -9,6 +9,7 @@ import (
 
 	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
 	"github.com/authzed/spicedb/pkg/tuple"
+	"github.com/authzed/spicedb/pkg/validationfile"
 	"github.com/jzelinskie/cobrautil/v2"
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
@@ -76,7 +77,7 @@ func importCmdFunc(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	var p decode.SchemaRelationships
+	var p validationfile.ValidationFile
 	if _, _, err := decoder(&p); err != nil {
 		return err
 	}
@@ -87,7 +88,7 @@ func importCmdFunc(cmd *cobra.Command, args []string) error {
 	}
 
 	if cobrautil.MustGetBool(cmd, "schema") {
-		if err := importSchema(cmd.Context(), client, p.Schema, prefix); err != nil {
+		if err := importSchema(cmd.Context(), client, p.Schema.Schema, prefix); err != nil {
 			return err
 		}
 	}
@@ -95,7 +96,7 @@ func importCmdFunc(cmd *cobra.Command, args []string) error {
 	if cobrautil.MustGetBool(cmd, "relationships") {
 		batchSize := cobrautil.MustGetInt(cmd, "batch-size")
 		workers := cobrautil.MustGetInt(cmd, "workers")
-		if err := importRelationships(cmd.Context(), client, p.Relationships, prefix, batchSize, workers); err != nil {
+		if err := importRelationships(cmd.Context(), client, p.Relationships.RelationshipsString, prefix, batchSize, workers); err != nil {
 			return err
 		}
 	}

--- a/internal/cmd/import_test.go
+++ b/internal/cmd/import_test.go
@@ -1,0 +1,62 @@
+package cmd
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/stretchr/testify/require"
+
+	"github.com/authzed/zed/internal/client"
+	zedtesting "github.com/authzed/zed/internal/testing"
+)
+
+var fullyConsistent = &v1.Consistency{Requirement: &v1.Consistency_FullyConsistent{FullyConsistent: true}}
+
+func TestImportCmdHappyPath(t *testing.T) {
+	require := require.New(t)
+	cmd := zedtesting.CreateTestCobraCommandWithFlagValue(t,
+		zedtesting.StringFlag{FlagName: "schema-definition-prefix"},
+		zedtesting.BoolFlag{FlagName: "schema", FlagValue: true},
+		zedtesting.BoolFlag{FlagName: "relationships", FlagValue: true},
+		zedtesting.IntFlag{FlagName: "batch-size", FlagValue: 100},
+		zedtesting.IntFlag{FlagName: "workers", FlagValue: 1},
+	)
+	f := filepath.Join("test", "happy-path-validation-file.yaml")
+
+	// Set up client
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	srv := zedtesting.NewTestServer(ctx, t)
+	go func() {
+		require.NoError(srv.Run(ctx))
+	}()
+	conn, err := srv.GRPCDialContext(ctx)
+	require.NoError(err)
+
+	originalClient := client.NewClient
+	defer func() {
+		client.NewClient = originalClient
+	}()
+
+	client.NewClient = zedtesting.ClientFromConn(conn)
+
+	c, err := zedtesting.ClientFromConn(conn)(cmd)
+	require.NoError(err)
+
+	// Run the import and assert we don't have errors
+	err = importCmdFunc(cmd, []string{f})
+	require.NoError(err)
+
+	// Run a check with full consistency to see whether the relationships
+	// and schema are written
+	resp, err := c.CheckPermission(ctx, &v1.CheckPermissionRequest{
+		Consistency: fullyConsistent,
+		Subject:     &v1.SubjectReference{Object: &v1.ObjectReference{ObjectType: "user", ObjectId: "1"}},
+		Permission:  "view",
+		Resource:    &v1.ObjectReference{ObjectType: "resource", ObjectId: "1"},
+	})
+	require.NoError(err)
+	require.Equal(resp.Permissionship, v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION)
+}

--- a/internal/cmd/test/happy-path-validation-file.yaml
+++ b/internal/cmd/test/happy-path-validation-file.yaml
@@ -1,0 +1,9 @@
+---
+schemaFile: "./happy-path-validation-schema.zed"
+relationships: >-
+  resource:1#user@user:1
+assertions:
+  assertTrue:
+    - "resource:1#user@user:1"
+  assertFalse:
+    - "resource:1#user@user:2"

--- a/internal/cmd/test/happy-path-validation-schema.zed
+++ b/internal/cmd/test/happy-path-validation-schema.zed
@@ -1,0 +1,6 @@
+definition user {}
+
+definition resource {
+  relation user: user
+  permission view = user
+}

--- a/internal/decode/decoder.go
+++ b/internal/decode/decoder.go
@@ -113,7 +113,10 @@ func unmarshalAsYAMLOrSchemaWithFile(data []byte, out interface{}, filename stri
 		if err := yaml.Unmarshal(data, out); err != nil {
 			return false, err
 		}
-		validationFile := out.(*validationfile.ValidationFile)
+		validationFile, ok := out.(*validationfile.ValidationFile)
+		if !ok {
+			return false, fmt.Errorf("could not cast unmarshalled file to validationfile")
+		}
 
 		// Need to join the original filepath with the requested filepath
 		// to construct the path to the referenced schema file.

